### PR TITLE
Update core dependencies and remove final traces of the Authlist feature

### PR DESCRIPTION
**Pull request by sebastian-aranda**
This PR makes several updates on the dependencies of the LOVE-manager application. A set of the ones with major updates is:

- Python 3.8 -> 3.11
- Django 3.2.25 -> 5.0.7
- channels 3.0.3 -> 4.1.0
- channels-redis 3.2.0 -> 4.2.0
- aioredis 1.3.1 -> 2.0.1
- autobahn 20.12.3 -> 23.6.2
- django-cors-headers 3.2.1 -> 4.4.0
- hiredis 1.0.1 -> 3.0.0
- hyperlink 19.0.0 -> 21.0.0
- incremental 17.5.0 -> 22.10.0
- packaging 20.3 -> 24.1
- pytz 2019.3 -> 2024.1
- PyYAML 5.4 -> 6.0.1
- txaio 20.1.1 -> 23.1.1
- zope.interface 4.7.2 -> 6.4

Also adjustments for deprecated features were done.

Finally, a couple traces of the Authlist feature were removed too. This was causing bunch of warnings on the startup of the service.

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -1,10 +1,9 @@
-FROM python:3.8.2-buster
+FROM python:3.11-bookworm
```
Comment (wvreeven) [inline docker/Dockerfile:1]: Are there plans to migrate the Dockerfiles to AlmaLinux9? `bookworm` sounds like a Debian derivative. TSSW uses AlmaLinux with Mamabaforge for Python and conda.

File: CHANGELOG.rst [hunk 2]
```diff
@@ -2,6 +2,11 @@
 Version History
 ===============
 
+v7.0.0
+------
+
+* Update core dependencies `<https://github.com/lsst-ts/LOVE-manager/pull/270>`_
```
Comment (wvreeven) [inline CHANGELOG.rst:8]: Looks like more happened, like remove Authlist support. Not sure if you want to mention that here as well?

**Review by wvreeven — APPROVED** [review]
LGTM. I left a few comments and questions for your consideration.

File: docker/Dockerfile (hunk 1 above)
Comment (sebastian-aranda) [reply docker/Dockerfile:1]: I checked with @tribeiro and this is not a priority yet, but I extended a ticket description I had for this purpose to state we expect to use AlmaLinux too :+1: [DM-44923](https://rubinobs.atlassian.net/browse/DM-44923).

Comment (wvreeven) [reply docker/Dockerfile:1]: Thanks for the info!

**Pull request merged**